backend/src/datasources/postApi.js

```diff
@@ -0,0 +1,44 @@
+const { RESTDataSource } = require('apollo-datasource-rest');
+
+class PostsAPI extends RESTDataSource {
```

Comment:
```suggestion
const { DataSource } = require('apollo-datasource-rest');

class PostsAPI extends DataSource {
```
You're not fetching from a remote REST API.